### PR TITLE
added user, protocol and reserved user filter options in Connection F…

### PR DESCRIPTION
**Pull request by ct-goranpalibrk**
…ilter Panel

Signed-off-by: CT\pgoran <goran.palibrk@comtrade.com>

column name and filter field name changed

Signed-off-by: CT\pgoran <goran.palibrk@comtrade.com>

no_user filter option removed from Connections Filter Panel

Signed-off-by: CT\pgoran <goran.palibrk@comtrade.com>

Brief description of the PR.
Added new fields(Last User, Protocol, Reserved Users) in Connections Filter Panel.

**Related Issue**
This PR is second part of PR's and issue which is solved by this, will be referenced in next PR.

**Description of the solution adopted**
New fields added in Connections Filter Panel, and new if clauses added in GwtKapuaConnectionsModelConverter.

**Screenshots**
If applicable, add screenshots to help explain your solution

**Any side note on the changes made**
Description of any other change that has been made, which is not directly linked to the issue resolution
[e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

**Comment by LeoNerdoG** [comment]
I have verified that this PR successfully resolves part of issue _**1857**_. @muros-ct @lorthirk  can you please check if the code and implementation as a whole is ok and merge?

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/model/GwtDeviceConnection.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2017 Eurotech and/or its affiliates and others
+ * Copyright (c) 2017, 2018 Eurotech and/or its affiliates and others
```
Comment (muros-ct) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/model/GwtDeviceConnection.java]: This is the only change. Please revert.

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/util/GwtKapuaDeviceModelConverter.java [hunk 2]
```diff
@@ -11,8 +11,9 @@
  *******************************************************************************/
 package org.eclipse.kapua.app.console.module.device.shared.util;
 
-import com.extjs.gxt.ui.client.Style.SortDir;
-import com.extjs.gxt.ui.client.data.PagingLoadConfig;
+import java.util.ArrayList;
+import java.util.List;
```
Comment (muros-ct) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/util/GwtKapuaDeviceModelConverter.java]: Please revert the order of imports as no changes were made.

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/util/GwtKapuaDeviceModelConverter.java [hunk 3]
```diff
@@ -70,6 +71,27 @@ public static DeviceConnectionQuery convertConnectionQuery(PagingLoadConfig load
         if (gwtDeviceConnectionQuery.getClientIP() != null && !gwtDeviceConnectionQuery.getClientIP().trim().isEmpty()) {
             predicate.and(new AttributePredicateImpl<String>(DeviceConnectionPredicates.CLIENT_IP, gwtDeviceConnectionQuery.getClientIP(), Operator.LIKE));
         }
+        if (gwtDeviceConnectionQuery.getUserName() != null && !gwtDeviceConnectionQuery.getUserName().trim().isEmpty()) {
+            predicate.and(new AttributePredicateImpl<String>(DeviceConnectionPredicates.USER_ID, gwtDeviceConnectionQuery.getUserName(), Operator.LIKE));
+        }
+
```
Comment (muros-ct) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/util/GwtKapuaDeviceModelConverter.java]: Are additional blank lines necessary here and in next if statements?

**Comment by ct-goranpalibrk** [comment]
@muros-ct PR has been changed according comments.

**Comment by ct-goranpalibrk** [comment]
@muros-ct Added one more thing which you requested and I forgot in last commit, and that 2 commits are squashed into this one.

**Comment by LeoNerdoG** [comment]
@muros-ct I have verified that this PR still successfully resolves issue _**1857**_ so it can be merged when ready.

**Pull request merged**